packages/orchestration/src/examples/send-anywhere.contract.js

```diff
@@ -33,19 +33,18 @@ const contract = async (
   zcf,
   privateArgs,
   zone,
-  { chainHub, orchestrateAll, zoeTools },
+  { atomicProvider, chainHub, orchestrate, orchestrateAll, vowTools, zoeTools },
 ) => {
-  const contractState = makeSharedStateRecord(
-    /** @type {{ account: OrchestrationAccount<any> | undefined }} */ {
-      localAccount: undefined,
-    },
-  );
-
   const creatorFacet = prepareChainHubAdmin(zone, chainHub);
 
+  const sharedLocalAccount = await atomicProvider.provideAsync(
+    'localAccount',
+    key => vowTools.when(orchestrate(key, {}, sharedFlows.makeLocalAccount)()),
+  );
```

Comment:
If I understand what's going on here, this likely does not work on restart, and is upgrade hazardous for other reasons.

Afaik, `provideAsync` lazily invokes the maker if the value is not found in the store (nor pending).

`orchestrate` is an exo definition that needs to be rewired during start for every subsequent incarnation. 

Given that the maker given to provide here contains the `orchestrate` call, and that it only happens once, the orchestrated flow will never be rewired in future incarnations, preventing the vat from upgrading.

Furthermore, the `await` here is for a non immediate promise. It is either dependent on network or on other vats, and would prevent the start result promise from immediately settling during the first crank. The only reason it works right now is because the first incarnation of the contract has this logic, and there is a limitation in zcf which cannot enforce single crank first start of contract (because the first zcf start is actually done in a second crank). If this was added as new logic in a second incarnation of the contract, the upgrade to that incarnation would fail.

I believe the following would work:

```suggestion
  const { makeLocalAccount } = orchestrateAll(sharedFlows, {});
  const sharedLocalAccount = zone.makeOnce(
    'localAccount',
    key => makeLocalAccount(),
  );
```

Then of course you still need to `await` the `sharedLocalAccount` inside flow, but there is just no way around that.

I imagine a slightly more reusable approach would be:

```js
  const orchAccountsProvider = zone.subZone('orchAccounts');
  const { makeOrchAccount } = orchestrateAll(sharedFlows, {});
  const provideOrchAccount = chain => orchAccountProvider.makeOnce(chain, makeOrchAccount);
```

Then `provideOrchAccount` can be given as an endowment. It's return value of course must still be awaited in flows.

Reply:
Great review! 🙏